app/containers/write.js

```diff
@@ -93,7 +94,38 @@ class writeScreen extends Component {
   componentDidUpdate (prevProps) {
     if (this.props.currentChannel !== prevProps.currentChannel) {
       this.focusInput()
-    }
+      var i = 0 
+      var flag = 0
+      for(i = 0; i < this.myArray.length; i++){
+        if (prevProps.currentChannel == this.myArray[i].name){
+          if (this.textInput.value != this.myArray[i].text){
+            window[this.props.currentChannel] = {
```

Comment:
Instead of saving the channel's input text on window could you send it through a redux reducer (maybe on keypress) that saves in the store? Then other things can know and respond to text changes in the input for that channel.